runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -340,12 +369,19 @@ public String toString() {
      * Verify that max payload is enforced for the specified limit.
      *
      * @param limit Max write limit.
+     * @return the serialized size of the payload
      */
-    public void checkMaxWriteSize(int limit) {
-        try (ILogData.SerializationHandle sh = this.getSerializedForm()) {
-            if (limit != 0 && getSizeEstimate() > limit) {
-                throw new WriteSizeException(getSizeEstimate(), limit);
-            }
+    public int checkMaxWriteSize(int limit) {
+        if (serializedCache == null) {
```

Comment:
when one small method throws two different exceptions it seems too much, can we throw the only WriteSizeException?

Reply:
A precondition check is required to throw IllegalStateException